frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -342,11 +345,100 @@ export const getTopologyGroupItems = (dc: K8sResourceKind, groups: Group[]): Gro
       });
     } else {
       const gIndex = _.findIndex(groups, { name: label });
-      groups[gIndex].nodes.push(uid);
+      if (!groups[gIndex].nodes.includes(uid)) {
+        groups[gIndex].nodes.push(uid);
+      }
     }
   });
   return groups;
 };
+/**
+ * Creates the operator backed services topology data
+ * @param resources
+ * @param transformBy
+ * @param installedOperators
+ * @param operatorBackedServiceKindMap
+ * @param topologyGraphAndNodeData
+ */
+const getOperatoarBackedServiceData = (
+  resources: TopologyDataResources,
+  transformBy: string[],
+  installedOperators: K8sResourceKind[],
+  operatorBackedServiceKindMap: OperatorBackedServiceKindMap,
+  topologyGraphAndNodeData: TopologyDataModel,
+): TopologyDataModel => {
+  const obsGroups = {};
+  const operatorMap = {};
+  let obsNodeItems = [];
+  const obsTopology = {};
+  const {
+    graph: { nodes, edges, groups },
+    topology,
+  } = topologyGraphAndNodeData;
+  if (transformBy.includes('deployments')) {
+    _.map(resources.deployments.data, (deployment) => {
+      const nodeResourceKind = _.get(deployment, 'metadata.ownerReferences[0].kind');
+      const ownerUid = _.get(deployment, 'metadata.ownerReferences[0].uid');
+      const appGroup = _.get(deployment, ['metadata', 'labels', 'app.kubernetes.io/part-of']);
+      let operator: K8sResourceKind = _.find(installedOperators, {
+        metadata: { uid: ownerUid },
+      }) as K8sResourceKind;
+
+      if (_.isEmpty(operator)) {
+        operator = operatorBackedServiceKindMap[nodeResourceKind];
+      }
```

Comment:
So we create `operatorBackedServiceKindMap` using the `installedOperators`, if operator is not present in `installedOperators` then it will not be present in `operatorBackedServiceKindMap`. Am I missing anything?

Reply:
we create operatorBackedServiceKindMap(CRD map) out of installedOperators, which can be used to identify the resources created by the crd. But when you install an operator, the operator creates a resource (eg: postgres-operator) which will have ownerReference of the installed operator(csv). We need to group that nodes create by operator & crd together.

Reply:
Based on a resource's owner reference, this will choose from `installedOperators` or its crds `operatorBackedServiceKindMap`

Reply:
Got it. Forgot about the case of the operator node :P Then it makes sense to look for the csv itself in the installed operators list.